packages/ra-ui-materialui/src/list/datagrid/Datagrid.tsx

```diff
@@ -117,9 +117,16 @@ const defaultBulkActionButtons = <BulkDeleteButton />;
  *     );
  * }
  */
-export const Datagrid: React.ForwardRefExoticComponent<
-    Omit<DatagridProps, 'ref'> & React.RefAttributes<HTMLTableElement>
-> = React.forwardRef<HTMLTableElement, DatagridProps>((props, ref) => {
+
+const fixedForwardRef = <T, P = {}>(
+    render: (props: P, ref: React.Ref<T>) => React.ReactNode
+): ((props: P & React.RefAttributes<T>) => React.ReactNode) =>
+    React.forwardRef(render) as any;
```

Comment:
Can you rename this to `genericForwardRef` (just like our `genericMemo`) and extract it in its own file directly under `src`? I have a feeling we're going to use it elsewhere